packages/theme-ui/src/provider.js

```diff
@@ -69,7 +37,11 @@ const RootProvider = ({ theme: propsTheme = {}, components, children }) => {
   const outer = useThemeUI()
   const [colorMode, setColorMode] = useColorState(propsTheme.initialColorMode)
   const [themeState, setThemeState] = useReducer(mergeState, propsTheme)
+
   const theme = applyColorMode(themeState, colorMode)
+  if (theme.useCustomProperties) {
+    theme.colors = toCustomProperties(theme.colors, 'colors')
```

Comment:
Currently this is still only converting the `colors` object to custom properties